staging/src/k8s.io/apiserver/pkg/server/options/authentication.go

```diff
@@ -233,7 +244,12 @@ func NewDelegatingAuthenticationOptions() *DelegatingAuthenticationOptions {
 		CacheTTL:   10 * time.Second,
 		ClientCert: ClientCertAuthenticationOptions{},
 		RequestHeader: RequestHeaderAuthenticationOptions{
-			UsernameHeaders:     []string{"x-remote-user"},
+			UsernameHeaders: []string{"x-remote-user"},
+			// we specifically don't default UID headers as these were introduced
+			// later and we don't want 3rd parties to be trusting the default headers
+			// before we can safely say that all KAS instances know they should
+			// remove them from an incoming request in its WithAuthentication handler
```

Comment:
note the specific release KAS handles this (1.32+) and / or link to the issue tracking the promotion to default (https://github.com/kubernetes/kubernetes/issues/126821)